applications/crossbar/src/modules/cb_global_provisioner_templates.erl

```diff
@@ -65,30 +65,26 @@ init() ->
 %% Add content types provided by this module
 %% @end
 %%--------------------------------------------------------------------
--spec content_types_provided(cb_context:context(), path_token(), path_token()) ->
-                                    crossbar_content_handlers().
-content_types_provided(Context, DocId, ?IMAGE_REQ) ->
-    case cb_context:req_verb(Context) of
-        ?HTTP_GET ->
-            case couch_mgr:open_doc(?WH_PROVISIONER_DB, DocId) of
-                {'error', _} -> Context;
-                {'ok', JObj} ->
-                    [Type, SubType] = binary:split(get_content_type(JObj), <<"/">>),
-                    lager:debug("found attachement of content type: ~s/~s~n", [Type, SubType]),
-                    cb_context:set_content_types_provided(Context, [{'to_binary', [{Type, SubType}]}])
-            end;
-        _ -> Context
+-spec content_types_provided(cb_context:context(), path_token(), path_token()) -> crossbar_content_handlers().
+-spec content_types_provided(cb_context:context(), http_method(), path_token(), path_token()) -> crossbar_content_handlers().
+content_types_provided(Context, PT1, PT2) ->
+    content_types_provided(Context, cb_context:req_verb(Context), PT1, PT2).
```

Comment:
I don't understand why you and @macpie mix up the arg order when cascading into higher arity functions! :) Please, no more...